### PR TITLE
Fixes example snippet in Result.swift

### PR DESCRIPTION
**Pull request by valeriyvan**
Fixes example snippet in Result.swift

### CODE REVIEW

File: stdlib/public/core/Result.swift [hunk 1]
```diff
@@ -131,7 +131,7 @@ public enum Result<Success, Failure: Error> {
   ///     do {
   ///         let value = try integerResult.get()
   ///         print("The value is \(value).")
-  ///     } catch error {
+  ///     } catch let error {
```
Comment (xwu) [inline stdlib/public/core/Result.swift:134]: Good catch. I think the example can be simplified though: there's no need to bind the error for the didactic purposes here.

```suggestion
  ///     } catch {
```

File: stdlib/public/core/Result.swift [hunk 2]
```diff
@@ -131,7 +131,7 @@ public enum Result<Success, Failure: Error> {
...
   ///         print("Error retrieving the value: \(error)")
```
Comment (xwu) [inline stdlib/public/core/Result.swift:135]:
```suggestion
  ///         print("Error retrieving the value.")
```

File: stdlib/public/core/Result.swift (hunk 1 above)
Comment (valeriyvan) [reply stdlib/public/core/Result.swift]: done

**Comment by xwu** [comment]
@swift-ci smoke test

**Review by natecook1000 — APPROVED** [review]

**Comment by CodaFi** [comment]
⛵

**Pull request merged**